spec/controllers/admin_request_controller_spec.rb

```diff
@@ -55,18 +55,21 @@
 
   describe 'when fully destroying a request' do
 
-    it 'expires the file cache for that request' do
-      info_request = info_requests(:badger_request)
-      expect(@controller).to receive(:expire_for_request).with(info_request)
-      get :destroy, { :id => info_request }
-    end
-
     it 'uses a different flash message to avoid trying to fetch a non existent user record' do
       info_request = info_requests(:external_request)
       post :destroy, { :id => info_request.id }
       expect(request.flash[:notice]).to include('external')
     end
 
+    it 'destroys associated incoming_message records' do
```

Comment:
The "can't modify frozen hash" error was being triggered when attempting to destroy requests with associated incoming messages